File: app/src/main/java/com/github/sewerina/meter_readings/ui/chart/ChartViewModel.java
package com.github.sewerina.meter_readings.ui.chart;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.github.sewerina.meter_readings.ReadingApp;
import com.github.sewerina.meter_readings.database.AppDao;
import com.github.sewerina.meter_readings.database.HomeEntity;
import com.github.sewerina.meter_readings.database.ReadingEntity;

import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

public class ChartViewModel extends ViewModel {
    private final MutableLiveData<List<ReadingEntity>> mReadingEntities = new MutableLiveData<>();
    private AppDao mDao;
    private final CompositeDisposable mDisposables = new CompositeDisposable();

    public ChartViewModel() {
        mDao = ReadingApp.mReadingDao;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        mDisposables.dispose();
    }

    public LiveData<List<ReadingEntity>> getReadings() {
        return mReadingEntities;
    }

    public void loadInChart(final HomeEntity currentHomeEntity) {
        Disposable subscribe = mDao.getReadingsForHomeRx(currentHomeEntity.id)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<List<ReadingEntity>>() {
                    @Override
                    public void accept(List<ReadingEntity> readingEntities) throws Exception {
                        mReadingEntities.postValue(readingEntities);
                    }
                });
        mDisposables.add(subscribe);
    }
}

File: app/src/main/java/com/github/sewerina/meter_readings/ui/chart/ChartViewModel.kt
package com.github.sewerina.meter_readings.ui.chart

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.github.sewerina.meter_readings.database.AppDao
import com.github.sewerina.meter_readings.database.HomeEntity
import com.github.sewerina.meter_readings.database.ReadingEntity
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class ChartViewModel @Inject constructor(
    var mDao: AppDao
) : ViewModel() {
    private val mReadingEntities = MutableLiveData<List<ReadingEntity>>()
    val readings: LiveData<List<ReadingEntity>>
        get() = mReadingEntities

    fun loadInChart(currentHomeEntity: HomeEntity) {
        viewModelScope.launch {
            val readingEntities = mDao.getReadingsForHome(currentHomeEntity.id)
            mReadingEntities.postValue(readingEntities)
        }
    }
}